jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/settings/component/binder/AbstractGridSettingsBinder.java

```diff
@@ -166,7 +171,7 @@ protected boolean isColumnSortOrderChanged(@Nullable List<? extends GridSortOrde
 
             DataGridSettings.SortOrder sSortOrder = settingsSortOrder.get(i);
 
-            if (!key.equals(sSortOrder.getKey())) {
+            if (key != null && !key.equals(sSortOrder.getKey())) {
```

Comment:
if `sSortOrder.getKey()` is nut null, then this code can be simplified to `!sSortOrder.getKey().equals(key)` or `java.util.Objects#equals`

Reply:
sort order key can be null, added @Nullable